mopidy/file/library.py

```diff
@@ -47,38 +44,39 @@ def browse(self, uri):
         result = []
         local_path = path.uri_to_path(uri)
 
-        if local_path == 'root':
+        # py-compat: Use str() to get a native string on both Py2/3
+        if str(local_path) == str('root'):
             return list(self._get_media_dirs_refs())
 
-        if not self._is_in_basedir(os.path.realpath(local_path)):
+        if not self._is_in_basedir(local_path):
             logger.warning(
                 'Rejected attempt to browse path (%s) outside dirs defined '
```

Comment:
Example of logging a uri with `%s`.

Reply:
We're logging an URI, not a path here, so I think we should be good.

Reply:
My concern was that a py2 uri is bytes, which may contain non-ASCII, which is mixed here with a unicode literal.

Reply:
I see that. URIs *should* be URI-encoded, and if so, only contains ASCII. Of course, theory does not equal practise.

One way to solve this is to just move ahead towards Py3, and get rid of Py2.